### PR TITLE
Perf e2e test app fragments

### PR DESCRIPTION
**Pull request by leotianlizhan**
b/208271664

The tests are run on Firebase Test Lab automatically. There are no validation however for screen traces. 

## Result
The instrumentation test will automatically scroll through the the laggy `RecyclerView` on a fragment, then move to the next fragment. 

There was also a change to the old activity screen trace test, to try to make the instrumentation test produce screen trace which didn't do it consistently before. 

## How to Test
Run instrumentation tests: `./gradlew :firebase-perf:e2e-app:connectedCheck -PfireperfBuildForAutopush`

### CODE REVIEW

**Review by jeremyjiang-dev — APPROVED** [review]
Great work!

File: firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceScreenTracesTest.java [hunk 1]
```diff
@@ -41,6 +48,16 @@
           /* initialTouchMode= */ false,
           /* launchActivity= */ true);
 
+  @After
+  public void pressHome_toTriggerSendScreenTrace() {
```
Comment (visumickey) [inline firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceScreenTracesTest.java:52]: Are we triggering an event sent for every app backgrounding? I thought we were dispatching events scheduled every 30 seconds managed by Firelog service. @jeremyjiang-dev

File: firebase-perf/e2e-app/src/main/res/navigation/mobile_navigation.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?> <!-- Copyright 2021 Google Inc. All Rights Reserved. -->
+<navigation xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/mobile_navigation"
+    app:startDestination="@+id/navigation_home">
+
+    <fragment
+        android:id="@+id/navigation_home"
```
Comment (visumickey) [inline firebase-perf/e2e-app/src/main/res/navigation/mobile_navigation.xml:9]: Can we name the fragments by which is a slow fragment and which is a fast fragment? Eg: can we name fragments as "Slow Fragment" and  "Fast fragment"? (this is incase if some fragments are slow and some of them are fast). If not, can we make fragments slow and fast?

**Review by visumickey — APPROVED** [review]

File: firebase-perf/e2e-app/e2e-app.gradle [hunk 3]
```diff
@@ -28,7 +28,7 @@ firebaseTestLab {
 }
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 31
```
[the remark this answers is not in the document]
Comment (leotianlizhan) [reply firebase-perf/e2e-app/e2e-app.gradle:31]: I am not sure what is causing the issue, but if I don't use 31 the app fails at building. Maybe a library I added.

File: firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceFragmentScreenTracesTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      currItemCount += 5;
+    }
+  }
+
+  private synchronized void blockUntilNavigationDone() throws InterruptedException {
+    wait();
+  }
+
+  private synchronized void notifyNavigationLock() {
+    notify();
```
[the remark this answers is not in the document]
Comment (leotianlizhan) [reply firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceFragmentScreenTracesTest.java:99]: Every object in Java has an intrinsic lock. `wait()` causes current thread to wait indefinitely until `notify()` is called on the same object.

File: firebase-perf/e2e-app/e2e-app.gradle [hunk 5]
```diff
@@ -52,7 +52,7 @@ android {
         // https://developer.android.com/studio/build/application-id.html
         applicationId "com.google.firebase.testing.fireperf"
 
-        minSdkVersion 16
+        minSdkVersion 18
```
[the remark this answers is not in the document]
Comment (leotianlizhan) [reply firebase-perf/e2e-app/e2e-app.gradle:55]: `androidx.test.uiautomator:uiautomator:2.2.0` requires API 18 at least, and that is being used to programmatically go to home screen during test.

File: firebase-perf/e2e-app/src/main/res/navigation/mobile_navigation.xml (hunk 2 above)
Comment (leotianlizhan) [reply firebase-perf/e2e-app/src/main/res/navigation/mobile_navigation.xml:9]: Done, made dashboard fragment into "FastFragment" and notification fragment into "SlowFragment"

**Review by visumickey — APPROVED** [review]

File: firebase-perf/e2e-app/e2e-app.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (visumickey) [reply firebase-perf/e2e-app/e2e-app.gradle:55]: Sounds good!

**Pull request merged**